sdk/keyvault/keyvault/src/secrets/index.ts

```diff
@@ -0,0 +1,470 @@
+   * @param secretName The name of the secret.
+   * @param [options] The optional parameters
+   * @returns Promise<Secret>
+   */
+  public async getSecret(
+    secretName: string,
+    options?: GetSecretOptions
+  ): Promise<Secret> {
+    const response = await this.client.getSecret(
+      this.vaultBaseUrl,
+      secretName,
+      options && options.version ? options.version : "",
+      options ? options.requestOptions : undefined
+    );
+    return this.getSecretFromSecretBundle(response);
+  }
+
+  /**
+   * The Get Deleted Secret operation returns the specified deleted secret along with its attributes.
+   * This operation requires the secrets/get permission.
+   * @summary Gets the specified deleted secret.
+   * @param secretName The name of the secret.
+   * @param [options] The optional parameters
+   * @returns Promise<DeletedSecret>
+   */
+  public async getDeletedSecret(
+    secretName: string,
+    options?: RequestOptionsBase
+  ): Promise<DeletedSecret> {
+    const response = await this.client.getDeletedSecret(this.vaultBaseUrl, secretName, options);
+    return this.getSecretFromSecretBundle(response);
+  }
+
+  /**
+   * The purge deleted secret operation removes the secret permanently, without the possibility of
+   * recovery. This operation can only be enabled on a soft-delete enabled vault. This operation
+   * requires the secrets/purge permission.
+   * @summary Permanently deletes the specified secret.
+   * @param secretName The name of the secret.
+   * @param [options] The optional parameters
+   * @returns Promise<void>
+   */
+  public async purgeDeletedSecret(secretName: string, options?: RequestOptionsBase): Promise<void> {
+    await this.client.purgeDeletedSecret(this.vaultBaseUrl, secretName, options);
+  }
+
+  /**
+   * Recovers the deleted secret in the specified vault. This operation can only be performed on a
+   * soft-delete enabled vault. This operation requires the secrets/recover permission.
+   * @summary Recovers the deleted secret to the latest version.
+   * @param secretName The name of the deleted secret.
+   * @param [options] The optional parameters
+   * @returns Promise<Secret>
+   */
+  public async recoverDeletedSecret(
+    secretName: string,
+    options?: RequestOptionsBase
+  ): Promise<Secret> {
+    const response = await this.client.recoverDeletedSecret(this.vaultBaseUrl, secretName, options);
+    return this.getSecretFromSecretBundle(response);
+  }
+
+  /**
+   * Requests that a backup of the specified secret be downloaded to the client. All versions of the
+   * secret will be downloaded. This operation requires the secrets/backup permission.
+   * @summary Backs up the specified secret.
+   * @param secretName The name of the secret.
+   * @param [options] The optional parameters
+   * @returns Promise<Uint8Array | undefined>
+   */
+  public async backupSecret(
+    secretName: string,
+    options?: RequestOptionsBase
+  ): Promise<Uint8Array | undefined> {
```

Comment:
>Promise<Uint8Array | undefined> [](start = 5, length = 31)

why is this | undefined?

Reply:
In coordination with @jonathandturner, let's address this through this issue: https://github.com/Azure/azure-sdk-for-js/issues/3030